glusterd2/volgen/store-utils.go

```diff
@@ -0,0 +1,91 @@
+package volgen
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+
+	"github.com/gluster/glusterd2/glusterd2/store"
+	"github.com/gluster/glusterd2/pkg/errors"
+
+	"github.com/coreos/etcd/clientv3"
+	log "github.com/sirupsen/logrus"
+)
+
+var (
+	volfilePrefix = "volfiles/"
+)
+
+type volfileData struct {
+	VolumeName string
+	Content    string
+}
+
+func save(name string, content string, volname string) error {
+	jsondata, err := json.Marshal(volfileData{VolumeName: volname, Content: content})
+	if err != nil {
+		log.WithField("error", err).Error("Failed to marshal the volfile data")
+		return err
+	}
+	_, err = store.Put(context.TODO(), volfilePrefix+name, string(jsondata))
+	return err
+}
+
+// GetVolfiles returns list of all Volfiles
+func GetVolfiles() ([]string, error) {
+	resp, e := store.Get(context.TODO(), volfilePrefix, clientv3.WithPrefix())
+	if e != nil {
+		return nil, e
+	}
+
+	volfiles := make([]string, len(resp.Kvs))
+
+	for i, kv := range resp.Kvs {
+		volFile := strings.TrimPrefix(string(kv.Key), "volfiles/")
+		volfiles[i] = volFile
+	}
+
+	return volfiles, nil
+}
+
+//GetVolfile return particular volfile info
+func GetVolfile(volfileID string) ([]byte, error) {
+	volfile := volfilePrefix + volfileID
+	resp, e := store.Get(context.TODO(), volfile, clientv3.WithPrefix())
+	if e != nil {
+		return []byte{}, errors.ErrFetchingVolfileContent
+	}
+	if len(resp.Kvs) == 0 {
+		return []byte{}, errors.ErrVolFileNotFound
+	}
+	var v volfileData
+	if e = json.Unmarshal(resp.Kvs[0].Value, &v); e != nil {
+		log.WithError(e).Error("Failed to unmarshal the data into volfiledata object")
+		return nil, e
+	}
+	return []byte(v.Content), nil
+}
+
+// DeleteVolumeVolfiles deletes all the Volfiles belonging to a Volume
+func DeleteVolumeVolfiles(volname string) error {
+	resp, e := store.Get(context.TODO(), volfilePrefix, clientv3.WithPrefix())
+	if e != nil {
+		return e
+	}
+
```

Comment:
Suggestion: If all volfiles belonging to a volume are under one prefix - `volfiles/<volname>`, you can delete all with one op usingWithPrefix()`

Reply:
good idea, rebalance(`rebalance/<volname>`) and gfproxy volfiles(`gfproxy-client/<volname>` and `gfproxyd/<volname>`) names are not in this format. I will change those daemons as well to use `<volname>...` as volfile id.

Reply:
I was hinting at just client and brick volfiles (per volume ones). The cluster volfiles aren't on per volume basis right ? It'll be global, isn't it ?

Reply:
yes. cluster volfiles are prefixed as `volfiles/gluster/`. Rebalance and gfproxy are also volume level volfiles(one per volume)